db/sqlc/migrations/000005_actions.up.sql

```diff
@@ -0,0 +1,52 @@
+CREATE TABLE IF NOT EXISTS actions(
+    -- The ID of the action.
+    id INTEGER PRIMARY KEY,
+
+    -- The session ID of the session that this action is associated with.
+    -- This may be null for actions that are not coupled to a session.
+    session_id BIGINT REFERENCES sessions(id) ON DELETE CASCADE,
+
+    -- The account ID of the account that this action is associated with.
+    -- This may be null for actions that are not coupled to an account.
+    account_id BIGINT REFERENCES accounts(id) ON DELETE CASCADE,
+
+    -- An ID derived from the macaroon used to perform the action.
+    macaroon_identifier BLOB,
```

Comment:
This could potentially be `BLOB NOT NULL` instead. We could potentially also add a constraint for the length of this column as well, since the unmarshaling of this requires exactly 4 bytes.
Note that even for actions where we don't set any macaroon, we pass a [4]byte array with each byte set to `0` bytes which will get written to this column. If this is not intentional, we should change this so that this gets set an empty array when being persisted, and then also handle that case during the unmarshaling.

Reply:
(edit: opening a prep PR)